### PR TITLE
Ensures reset of ir.config_params into target database when db are co…

### LINKED ISSUE #25: copydb --reset-uuid + enterprise contract
reset uuid in target

### PR DESCRIPTION
**Pull request by lmignon**
This will prevent conflicts between databases (db.uuid, db.secret ...)

Closes #25 
Progresses #26

### CODE REVIEW

File: click_odoo_contrib/_dbutils.py [hunk 1]
```diff
@@ -47,3 +47,17 @@ def db_management_enabled():
...
+    Reset config parameters to default value. This is useful to avoid
+    conflicts between databases on copy or restore
+    (dbuuid, ...)
+    """
+    with OdooEnvironment(dbname) as env:
+        try:
+            env["ir.config_parameter"].init(force=True)
+        except TypeError:
+            # odoo < 10
+            env.registry("ir.config_parameter").init(env.cr, force=True)
```
Comment (sbidoul) [inline click_odoo_contrib/_dbutils.py]: @lmignon a check on `odoo.release.version_info` would be more robust than catching `TypeError`.

File: tests/test_copydb.py [hunk 2]
```diff
@@ -110,15 +129,18 @@ def test_copydb_template_not_exists_target_exists():
         _dropdb(TEST_DBNAME_NEW)
 
 
-def test_copydb_no_source_filestore(pgdb):
+def test_copydb_no_source_filestore(odoodb):
     filestore_dir_new = odoo.tools.config.filestore(TEST_DBNAME_NEW)
+    filestore_dir_original = odoo.tools.config.filestore(odoodb)
+    if os.path.exists(filestore_dir_original):
+        shutil.rmtree(filestore_dir_original)
```
Comment (sbidoul) [inline tests/test_copydb.py]: Do you remember why this rmtree was necessary?

**Comment by rousseldenis** [comment]
@lmignon

File: tests/test_copydb.py (hunk 2 above)
Comment (lmignon) [reply tests/test_copydb.py:136]: We now use `odoodb` fixture in place of `pgdb` (We need a properly initialized Odoo db to check if settings are reset) . Therefore a filestore is created by default in odoo => 10.

File: click_odoo_contrib/_dbutils.py (hunk 1 above)
Comment (lmignon) [reply click_odoo_contrib/_dbutils.py:74]: done

**Comment by lmignon** [comment]
@sbidoul Requested changes implemented.

File: tests/test_copydb.py [hunk 3]
```diff
@@ -4,16 +4,18 @@
...
 from click.testing import CliRunner
-from click_odoo import odoo
+from click_odoo import OdooEnvironment, odoo
 
 from click_odoo_contrib._dbutils import db_exists
 from click_odoo_contrib.copydb import main
 
 TEST_DBNAME = "click-odoo-contrib-testcopydb"
 TEST_DBNAME_NEW = "click-odoo-contrib-testcopydb-new"
+_DEFAULT_IR_CONFIG_PARAMETERS = ["database.uuid", "database.create_date"]
```
Comment (sbidoul) [inline tests/test_copydb.py:18]: Do we want to handle the other database keys?

```
 database.create_date
 database.secret
 database.uuid
 database.enterprise_code
 database.expiration_date
 database.expiration_reason
```

Comment (lmignon) [reply tests/test_copydb.py:18]: This list is not the same into all the odoo versions... 
I maybe should reuse the definition provided by odoo:
https://github.com/odoo/odoo/blob/13.0/odoo/addons/base/models/ir_config_parameter.py#L18
```python
_default_parameters = {
    "database.secret": lambda: str(uuid.uuid4()),
    "database.uuid": lambda: str(uuid.uuid1()),
    "database.create_date": fields.Datetime.now,
    "web.base.url": lambda: "http://localhost:%s" % config.get('http_port'),
    "base.login_cooldown_after": lambda: 10,
    "base.login_cooldown_duration": lambda: 60,
}
```
https://github.com/odoo/odoo/blob/12.0/odoo/addons/base/models/ir_config_parameter.py#L18
```python
_default_parameters = {
    "database.secret": lambda: pycompat.text_type(uuid.uuid4()),
    "database.uuid": lambda: pycompat.text_type(uuid.uuid1()),
    "database.create_date": fields.Datetime.now,
    "web.base.url": lambda: "http://localhost:%s" % config.get('http_port'),
    "base.login_cooldown_after": lambda: 10,
    "base.login_cooldown_duration": lambda: 60,
}
```
https://github.com/odoo/odoo/blob/11.0/odoo/addons/base/ir/ir_config_parameter.py#L18
```python
_default_parameters = {
    "database.secret": lambda: pycompat.text_type(uuid.uuid4()),
    "database.uuid": lambda: pycompat.text_type(uuid.uuid1()),
    "database.create_date": fields.Datetime.now,
    "web.base.url": lambda: "http://localhost:%s" % config.get('http_port'),
}
```
https://github.com/odoo/odoo/blob/10.0/odoo/addons/base/ir/ir_config_parameter.py#L16
```python
_default_parameters = {
    "database.secret": lambda: (str(uuid.uuid4()), ['base.group_erp_manager']),
    "database.uuid": lambda: (str(uuid.uuid1()), []),
    "database.create_date": lambda: (fields.Datetime.now(), ['base.group_user']),
    "web.base.url": lambda: ("http://localhost:%s" % config.get('xmlrpc_port'), []),
}
```
https://github.com/odoo/odoo/blob/9.0/openerp/addons/base/ir/ir_config_parameter.py#L20
```python
_default_parameters = {
    "database.secret": lambda: (str(uuid.uuid4()), ['base.group_erp_manager']),
    "database.uuid": lambda: (str(uuid.uuid1()), []),
    "database.create_date": lambda: (datetime.datetime.now().strftime(misc.DEFAULT_SERVER_DATETIME_FORMAT), ['base.group_user']),
    "web.base.url": lambda: ("http://localhost:%s" % config.get('xmlrpc_port'), []),
}
```
https://github.com/odoo/odoo/blob/8.0/openerp/addons/base/ir/ir_config_parameter.py#L38
```python
_default_parameters = {
    "database.uuid": lambda: (str(uuid.uuid1()), []),
    "database.create_date": lambda: (datetime.datetime.now().strftime(misc.DEFAULT_SERVER_DATETIME_FORMAT), ['base.group_user']),
    "web.base.url": lambda: ("http://localhost:%s" % config.get('xmlrpc_port'), []),
}
```

Comment (lmignon) [reply tests/test_copydb.py:18]: Bu default `database.enterprise_code`,  `database.expiration_date` and  `database.expiration_reason` are never reset. We should maybe do it.

File: tests/test_copydb.py [hunk 4]
```diff
@@ -81,10 +81,28 @@ def ir_config_param_test_values(odoodb):
...
+                env.cr.execute(
+                    """
+                INSERT INTO
+                    ir_config_parameter
+                    (key, value)
+                VALUES
+                    (%s, 'test value')
+                """,
+                    (key,),
+                )
```
Comment (sbidoul) [inline tests/test_copydb.py:105]: nit: with `insert into ... on conflict (key) do update set value=%s` that can be done in one statement.

File: click_odoo_contrib/_dbutils.py [hunk 5]
```diff
@@ -61,3 +61,34 @@ def advisory_lock(cr, name):
...
+        WHERE key = 'database.enterprise_code';
+
+        UPDATE ir_config_parameter
+        SET value = 'copy'
+        WHERE key = 'database.expiration_reason'
+        AND value != 'demo';
+
+        UPDATE ir_config_parameter
+        SET value = CURRENT_DATE + INTERVAL '2 month'
+        WHERE key = 'database.expiration_date';
```
Comment (sbidoul) [inline click_odoo_contrib/_dbutils.py:91]: Any specific reason to change these keys and not delete them?

Comment (lmignon) [reply click_odoo_contrib/_dbutils.py:91]: Not really. if enterprise addons are installed I prefer to keep the related keys.

**Pull request merged**